tests/debugging/exploration/debugger.py

```diff
@@ -0,0 +1,226 @@
+import os
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Tuple
+from warnings import warn
+
+from ddtrace.debugging._config import config
+from ddtrace.debugging._debugger import Debugger
+from ddtrace.debugging._debugger import DebuggerModuleWatchdog
+from ddtrace.debugging._encoding import SnapshotJsonEncoder
+from ddtrace.debugging._function.discovery import FunctionDiscovery
+from ddtrace.debugging._probe.model import Probe
+from ddtrace.debugging._probe.poller import ProbePollerEvent
+from ddtrace.debugging._snapshot.collector import SnapshotCollector
+from ddtrace.debugging._snapshot.collector import SnapshotContext
+from ddtrace.debugging._snapshot.model import Snapshot
+from ddtrace.internal.compat import ExcInfoType
+from ddtrace.internal.module import origin
+from ddtrace.internal.utils.formats import asbool
+
+
+try:
+    COLS, _ = os.get_terminal_size()
+except Exception:
+    COLS = 80
+CWD = os.path.abspath(os.getcwd())
+TESTS = os.path.join(CWD, "test")
+try:
+    _ = os.getenv("VIRTUAL_ENV")
+    VENV = os.path.abspath(_) if _ is not None else None
+except TypeError:
+    warn(
+        "No virtual environment detected. Running without a virtual environment active might cause exploration tests "
+        "to instrument more than intended."
+    )
+    VENV = None
+
+RUN_MODULE = False  # sys.argv[:1] == ["-m"]
+ENCODE = asbool(os.getenv("DD_DEBUGGER_EXPL_ENCODE", True))
+
+
+class ModuleCollector(DebuggerModuleWatchdog):
+    def __init__(self, *args, **kwargs):
+        super(ModuleCollector, self).__init__(*args, **kwargs)
+
+    def on_collect(self, discovery):
+        raise NotImplementedError()
+
+    def after_import(self, module):
+        o = origin(module)
+        if o.startswith(CWD) and not o.startswith(TESTS) and (VENV is None or not o.startswith(VENV)):
```

Comment:
I wasn't sure where in the code the specific module we'd be interested in is instrumented for the debugger. A comment here would be helpful.